moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -700,10 +700,11 @@ void PlanningSceneMonitor::includeAttachedBodiesInOctree()
   boost::recursive_mutex::scoped_lock _(shape_handles_lock_);
 
   // clear information about any attached body, without refering to the AttachedBody* ptr (could be invalid)
-  for (AttachedBodyShapeHandles::iterator it = attached_body_shape_handles_.begin();
-       it != attached_body_shape_handles_.end(); ++it)
-    for (std::size_t k = 0; k < it->second.size(); ++k)
-      octomap_monitor_->forgetShape(it->second[k].first);
+  for (std::pair<const robot_state::AttachedBody* const,
```

Comment:
It can stay like this, but this is one of the few times that I'd prefer `auto`, when the type itself takes up two lines and the variable only lasts for three lines.

Reply:
I have the same feeling. Maybe it's better to make some additional `typedef`s to simplify this.